shield_with_impacts.gdshader

```diff
@@ -1,42 +1,69 @@
 shader_type spatial;
 
-render_mode unshaded;
+render_mode unshaded, cull_disabled;
 
 group_uniforms shield_parameters;
-uniform vec4 color : source_color;
+
+/** Color to use when rendering the shield */
+uniform vec4 color: source_color;
+
+/** How sharp the shield's outline is */
 uniform float fresnel_sharpness = 1.0;
 
+/** Distance to extend the shield out from each vertex */
+uniform float extend_distance = 0.0;
+
 group_uniforms impact_parameters;
-// this defines how many simultaneous impacts the shader will track
+
+/**
+ * How many impacts can be tracked by the shader.
+ * This should match the length of `impact_points` array that is passed in.
+ */
 const int impacts_tracked = 16;
+
 // information about impact locations (xyz) and their age (w)
+/**
+ * List of impact locations (xyz) and their age (w).
+ * Age must be updated outside of the shader and passed in each frame.
+ */
 uniform vec4 impact_points[impacts_tracked];
-// determines the size of the impacts
-// note that you could pass in an array here as well to determine impact size on a per-impact basis
-uniform float impact_size;
-// determines how sharply to curve the impacts expansion and fading
+
+/**
+ * Size of the impacts.
+ * This is used to determine how far the impact wave will expand.
+ *
+ * Possible improvement: switch this to an array to change the size of each impact.
+ */
+uniform float impact_size = 0.25;
+
+/** How sharply the impact wave will expand and fade */
```

Comment:
If you wrap `uniform` comments in `/** */`, they show up in the editor when you hover over them:

<img width="620" alt="Screenshot 2025-01-04 at 2 42 47 PM" src="https://github.com/user-attachments/assets/1f93f6a6-3025-432b-870f-119640f0b58f" />